WordPress/Sniffs/Theme/NonPrintableCheckSniff.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of non-printable characters in themes
+ *
+ * @link    https://make.wordpress.org/themes/handbook/review/required/theme-check-plugin/#info
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NonPrintableCheckSniff extends WordPress_Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+		'JS',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return ( array_merge( PHP_CodeSniffer_Tokens::$commentTokens, PHP_CodeSniffer_Tokens::$emptyTokens, PHP_CodeSniffer_Tokens::$stringTokens, array(
+			T_STRING,
+			T_INLINE_HTML,
+		) ) );
+
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile   The file being scanned.
+	 * @param int                  $stackPtr    The position of the current token in
+	 *                                          the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens   = $phpcsFile->getTokens();
+		$token    = $tokens[ $stackPtr ];
+		$filename = $phpcsFile->getFilename();
+		if ( preg_match( '/[\x00-\x08\x0B-\x0C\x0E-\x1F\x80-\xFF]/', $token['content'], $matches ) ) {
+			$warning = 'Non-printable characters were found in the ' . $filename . ' file. You may want to check this file for errors.';
+			$phpcsFile->addWarning( $warning, $stackPtr, 'NonPrintableCheck' );
```

Comment:
I'm also wondering if - when PHPCS finds a bad character - we should even bother checking the tokens in the rest of the file or if we should just skip to the end `return $phpcsFile->numTokens;` ?

This might give issues with the unit tests, but we can work around that with a `public $skip_to_end = true;` class property combined with a `// @codingStandardsChangeSetting WordPress.Theme.NonPrintableCheck skip_to_end false` annotation if it would be decided this is the way to go.